interp/iostream.go

```diff
@@ -0,0 +1,209 @@
+package interp
+
+// I/O streams are interfaces which allow file redirects and command pipelines to be treated
+// equivalently.
+
+import (
+	"bufio"
+	"errors"
+	"io"
+	"os/exec"
+	"syscall"
+)
+
+const (
+	notClosedExitCode = -127.0
```

Comment:
This is weird as a float. Let's make it -127 integer (shouldn't matter as an untyped constant anyway).

Edit: actually, I think we should avoid this altogether -- see suggestion below.